### PR TITLE
Strassen's Matrix Multiplication Method

### PR DESCRIPTION
**Pull request by Tiger-Sharks**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: divide_and_conquer/strassens_matrix_mulitplication.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Code Contributed by: Tiger Sharks
+
+*/
+
+#include <iostream>
+using namespace std;
+
+int main()
+{
+    int m1[2][2], m2[2][2], res[2][2], x, y;
```
Comment (krishnacx) [inline divide_and_conquer/strassens_matrix_mulitplication.cpp:33]: Does this apply to matrix of size  2*2

Comment (Tiger-Sharks) [reply divide_and_conquer/strassens_matrix_mulitplication.cpp:33]: > Does this apply to matrix of size 2*2

Yes, as of now, it applies to matrices of size 2*2, later on planning to do it for other even sized matrices.
As this multiplication technique applies only to matrices of even size.

File: divide_and_conquer/strassens_matrix_mulitplication.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            |   r+s    p+t-r-v|  
+5) Then, we ran a loop in order to print the resultant matrix
+
+
+Code Contributed by: Tiger Sharks
+
+*/
+
+#include <iostream>
+using namespace std;
```
Comment (Panquesito7) [inline divide_and_conquer/strassens_matrix_mulitplication.cpp:29]: This is discouraged. Please strictly use `std::` for each member.

File: divide_and_conquer/strassens_matrix_mulitplication.cpp (hunk 1 above)
Comment (Panquesito7) [inline divide_and_conquer/strassens_matrix_mulitplication.cpp:33]: To register matrices, you should use C++-style vectors, which could be either `std::vector` (requires the `vector` library) or `std::array` (requires the `array` library). You'll need to create a vector inside a vector.

File: divide_and_conquer/strassens_matrix_mulitplication.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    cout << endl << "Enter the Elements for Matrix 2:" << endl;
+    for (x = 0; x < 2; x++)
+    {
+        for (y = 0; y < 2; y++)
+        {
+            cin >> m2[x][y];
+        }
+    }
+
+    p = (m1[0][0] + m1[1][1]) * (m2[0][0] + m2[1][1]);
```
Comment (Panquesito7) [inline divide_and_conquer/strassens_matrix_mulitplication.cpp:54]: Separate the code into documented functions and structures if needed as well as add self-test cases using `assert` (requires the `cassert` library/header).

File: divide_and_conquer/strassens_matrix_mulitplication.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        cout << endl;
+        for (y = 0; y < 2; y++)
+        {
+            cout << res[x][y] << "\t";
+        }
+    }
+    cout << endl;
+
+    return 0;
+}
```
Comment (Panquesito7) [inline divide_and_conquer/strassens_matrix_mulitplication.cpp:79]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

File: divide_and_conquer/strassens_matrix_mulitplication.cpp (hunk 1 above)
Comment (Tiger-Sharks) [reply divide_and_conquer/strassens_matrix_mulitplication.cpp:33]: Actually, I am not familiar with style vectors, never used them, and the existing method is the method which I am comfortable with...😅

File: divide_and_conquer/strassens_matrix_mulitplication.cpp (hunk 4 above)
Comment (Tiger-Sharks) [reply divide_and_conquer/strassens_matrix_mulitplication.cpp:79]: Okay. Didn't had an idea for that. Thanks for the advice.

File: divide_and_conquer/strassens_matrix_mulitplication.cpp (hunk 1 above)
Comment (Panquesito7) [reply divide_and_conquer/strassens_matrix_mulitplication.cpp:33]: Feel free to come and discuss that in the [Discord](https://the-algorithms.com/discord) server. I could send you a few resources to learn about it. This is the format we use in our repositories.

Comment (Tiger-Sharks) [reply divide_and_conquer/strassens_matrix_mulitplication.cpp:33]: Okay. I refer some resources and do the needful ASAP...

Comment (Tiger-Sharks) [reply divide_and_conquer/strassens_matrix_mulitplication.cpp]: I made the required changes and reuploaded the code. Do check it out.

**Comment by Panquesito7** [comment]
Please check some of my comments above. I think that's some of the missing things.

**Pull request closed without merging**